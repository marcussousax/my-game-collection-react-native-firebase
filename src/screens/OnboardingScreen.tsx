import React from 'react'
import AppIntroSlider from 'react-native-app-intro-slider'
import { Image, StyleSheet, Text, View } from 'react-native'
import { StackScreenProps } from '@react-navigation/stack'

const data = [
    {
        title: 'Welcome',
        text: 'to my React Native app\n',
        image: require('../assets/bg-onboarding-welcome.png'),
        bg: '#fff',
        color: '#000'
    },
    {
        title: 'Share some Feedbacks',
        text: 'a "bare" React Native CRUD\n using Typescript and Firebase.',
        image: require('../assets/bg-onboarding-step2.png'),
        bg: '#fff'
    },
    {
        title: 'Enjoy!',
        text: 'this a studying-purpose project.\n feel free to test it and give some feedback.',
        image: require('../assets/bg-onboarding-enjoy.png'),
        bg: '#fff'
    }
]

type Item = typeof data[0]

type AuthStackParamList = {
    LoginScreen: undefined
    OnboardingScreen: undefined
}

const OnboardingScreen = ({
    navigation
}: StackScreenProps<AuthStackParamList>) => {
    const keyExtractor = (item: Item) => item.title

    const renderItem = ({ item }: { item: Item }) => {
        return (
            <View
                style={[
                    styles.slide,
                    {
                        backgroundColor: item.bg,
                        flex: 1
                    }
                ]}
            >
                <Text style={styles.title}>{item.title}</Text>
                <Image source={item.image} style={styles.image} />
                <Text style={styles.text}>{item.text}</Text>
            </View>
        )
    }

    return (
        <AppIntroSlider
            keyExtractor={keyExtractor}
            renderItem={renderItem}
            data={data}
            onDone={() => navigation.navigate('LoginScreen')}
            onSkip={() => navigation.navigate('LoginScreen')}
            showSkipButton={true}
            dotStyle={{ backgroundColor: '#C4C4C4' }}
            activeDotStyle={{ backgroundColor: '#43DFA8' }}
            renderDoneButton={() => (
                <Text
                    style={{
                        fontWeight: '600',
                        marginVertical: 12,
                        color: '#C4C4C4'
                    }}
                >
                    done
                </Text>
            )}
            renderNextButton={() => (
                <Text
                    style={{
                        fontWeight: '600',
                        marginVertical: 12,
                        color: '#C4C4C4'
                    }}
                >
                    next
                </Text>
            )}
            renderSkipButton={() => (
                <Text
                    style={{
                        fontWeight: '600',
                        marginVertical: 12,
                        color: '#C4C4C4'
                    }}
                >
                    skip
                </Text>
            )}
        />
    )
}

export default OnboardingScreen

const styles = StyleSheet.create({
    slide: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'blue'
    },
    image: {
        width: 320,
        height: 320,
        marginVertical: 32,
        borderRadius: 32
    },
    text: {
        color: '#000',
        textAlign: 'center',
        fontSize: 19
    },
    title: {
        fontSize: 21,
        color: '#43DFA8',
        fontWeight: 'bold',
        textAlign: 'center'
    }
})
