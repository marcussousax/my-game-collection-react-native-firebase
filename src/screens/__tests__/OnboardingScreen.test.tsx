import 'react-native'
import React from 'react'
import OnboardingScreen from '../OnboardingScreen'
import TestRenderer from 'react-test-renderer'

const testProps = () => ({
    navigation: {
        navigate: jest.fn(),
        route: jest.fn()
    }
})

describe('<OnboardingScreen>', () => {
    it('it renders correctly', () => {
        const comp = TestRenderer.create(
            <OnboardingScreen {...testProps} />
        ).toJSON()
        expect(comp).toMatchSnapshot()
    })
})
