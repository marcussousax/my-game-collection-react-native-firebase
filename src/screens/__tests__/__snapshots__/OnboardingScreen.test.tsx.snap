// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OnboardingScreen> it renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    bottomButton={false}
    bounces={false}
    data={
      Array [
        Object {
          "bg": "#fff",
          "color": "#000",
          "image": Object {
            "testUri": "../../../src/assets/bg-onboarding-welcome.png",
          },
          "text": "to my React Native app
",
          "title": "Welcome",
        },
        Object {
          "bg": "#fff",
          "image": Object {
            "testUri": "../../../src/assets/bg-onboarding-step2.png",
          },
          "text": "a \\"bare\\" React Native CRUD
 using Typescript and Firebase.",
          "title": "Share some Feedbacks",
        },
        Object {
          "bg": "#fff",
          "image": Object {
            "testUri": "../../../src/assets/bg-onboarding-enjoy.png",
          },
          "text": "this a studying-purpose project.
 feel free to test it and give some feedback.",
          "title": "Enjoy!",
        },
      ]
    }
    disableVirtualization={false}
    dotClickEnabled={true}
    extraData={1}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={3}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onDone={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onSkip={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderDoneButton={[Function]}
    renderItem={[Function]}
    renderNextButton={[Function]}
    renderSkipButton={[Function]}
    scrollEventThrottle={50}
    showDoneButton={true}
    showNextButton={true}
    showPrevButton={false}
    showSkipButton={true}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "blue",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#fff",
                  "flex": 1,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#43DFA8",
                  "fontSize": 21,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Welcome
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/bg-onboarding-welcome.png",
                }
              }
              style={
                Object {
                  "borderRadius": 32,
                  "height": 320,
                  "marginVertical": 32,
                  "width": 320,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 19,
                  "textAlign": "center",
                }
              }
            >
              to my React Native app

            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "blue",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#fff",
                  "flex": 1,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#43DFA8",
                  "fontSize": 21,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Share some Feedbacks
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/bg-onboarding-step2.png",
                }
              }
              style={
                Object {
                  "borderRadius": 32,
                  "height": 320,
                  "marginVertical": 32,
                  "width": 320,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 19,
                  "textAlign": "center",
                }
              }
            >
              a "bare" React Native CRUD
 using Typescript and Firebase.
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "blue",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#fff",
                  "flex": 1,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#43DFA8",
                  "fontSize": 21,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Enjoy!
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/bg-onboarding-enjoy.png",
                }
              }
              style={
                Object {
                  "borderRadius": 32,
                  "height": 320,
                  "marginVertical": 32,
                  "width": 320,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 19,
                  "textAlign": "center",
                }
              }
            >
              this a studying-purpose project.
 feel free to test it and give some feedback.
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "bottom": 16,
        "justifyContent": "center",
        "left": 16,
        "position": "absolute",
        "right": 16,
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 16,
            "justifyContent": "center",
            "margin": 16,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#43DFA8",
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 10,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#C4C4C4",
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 10,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#C4C4C4",
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 10,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#C4C4C4",
                "fontWeight": "600",
                "marginVertical": 12,
              }
            }
          >
            next
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "left": 0,
            "position": "absolute",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#C4C4C4",
                "fontWeight": "600",
                "marginVertical": 12,
              }
            }
          >
            skip
          </Text>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;
